jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -418,21 +412,24 @@ private void registerRestoreEnvVariables() {
 
 						while ((entry = envFileReader.readLine()) != null) {
 							if (!entry.isBlank()) {
-								String entrySplit[] = entry.split("="); //$NON-NLS-1$
+								// Only split into 2 (max) allow "=" to be contained in the value.
+								String entrySplit[] = entry.split("=", 2); //$NON-NLS-1$
 								if (entrySplit.length != 2) {
 									throw new IllegalArgumentException(
 											"Env File entry is not in the correct format: [envVarName]=[envVarVal]: "
 													+ entry);
 								}
 
 								String name = entrySplit[0];
-								String value = entrySplit[1];
-
-								if (oldTheUnmodifiableEnvironment.containsKey(name)) {
-									throw new IllegalArgumentException(
-											"Env file entry cannot modifiy a pre-existing entry: " + entry);
+								String newValue = entrySplit[1];
+								String oldValue = oldTheUnmodifiableEnvironment.get(name);
+								if (oldValue != null) {
+									if (!Objects.equals(oldValue, newValue)) {
+										System.err.println(String.format("WARNING: Env file entry cannot modifiy a pre-existing entry key %s:  %s != %s", name, oldValue, newValue));
```

Comment:
I'm hesitant about continuing in this case as it leaves the system in an inconsistent state.  Think about having an old invalid password encoded and continuing causing the account to be locked (we've had this kind of problem in our CI systems in the past, not with this feature of course).

Failing for an inconsistent env is a much safer fall back than continuing.

Reply:
> Failing for an inconsistent env is a much safer fall back than continuing.

Sure, we can go back to failure.  I was recollecting the conversation we had in https://github.com/eclipse-openj9/openj9/issues/13545#issuecomment-926868842

Here we were discussing a warning message.  I did not interpret that conversation to mean that we would fail restore outright.  But if we feel that is the place to start we can see how troublesome that is in practice.

Note that the existing code and my suggested change ignore my security concerns because they include the value in the message which may be sensitive.  I can update the the PR to go back to failing restore and only include the key in the message.

Reply:
I updated the PR to cause an IllegalArgumentException again.  I changed it to collect up all the invalid keys and include all of them in the error message.  It is more convenient to know all the invalid keys at once instead of fixing one only to have the next one found repetitively.